From bcc10b22e52bed1f3b02c46cb0122d284be22d16 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=EB=A7=88=EB=88=84=EC=97=98?= <nalla@hamal.uberspace.de>
Date: Wed, 17 Jun 2015 09:30:41 +0200
Subject: [PATCH 20/26] path-conversion: Introduce ability to switch off
 conversion.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

When calling windows native apps from MSYS2, the runtime tries to convert
commandline arguments by a specific set of rules. See [MinGW wiki]
(http://www.mingw.org/wiki/Posix_path_conversion).

If the user does not want that behavior on a big scale, e.g. inside a bash
script, the user can now set the the environment variable
`MSYS_NO_PATHCONV` when calling native windows commands.

Signed-off-by: 마누엘 <nalla@hamal.uberspace.de>
---
 winsup/cygwin/msys2_path_conv.cc | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/winsup/cygwin/msys2_path_conv.cc b/winsup/cygwin/msys2_path_conv.cc
index 3cd3f6e..730300e 100644
--- a/winsup/cygwin/msys2_path_conv.cc
+++ b/winsup/cygwin/msys2_path_conv.cc
@@ -343,6 +343,14 @@ skip_p2w:
     }
 
     /*
+     * Skip path mangling when environment indicates it.
+     */
+    const char *no_pathconv = getenv ("MSYS2_NO_PATHCONV");
+
+    if (no_pathconv)
+      goto skip_p2w;
+
+    /*
      * Prevent Git's :file.txt and :/message syntax from beeing modified.
      */
     if (*it == ':')
-- 
2.5.2

